runtime/compiler/control/CompilationThread.cpp

```diff
@@ -7515,8 +7605,17 @@ TR::CompilationInfoPerThreadBase::preCompilationTasks(J9VMThread * vmThread,
             && !_jitConfig->inlineFieldWatches;
          }
 
-      bool sharedClassTest = eligibleForRelocatableCompile &&
-                             !TR::Options::getAOTCmdLineOptions()->getOption(TR_NoStoreAOT);
+#if defined(J9VM_OPT_CRIU_SUPPORT)
+      _compInfo.acquireCompMonitor(vmThread);
+      bool checkpointInProgress = _compInfo.isCheckpointInProgress();
```

Comment:
After storing something in `checkpointInProgress` its value can change when setting `sharedClassTest` below. Does that have any correctness implications?

Reply:
No because this value is only used to not do an AOT compile for preemptive compilations. 

At this point, the compilation is already in progress so the only meaningful change in value would be to indicate that the checkpoint should be aborted, which at the moment can only happen because of shutdown. As such, the normal coordination of comp threads during shutdown apply.